regparser/tree/depth/derive.py

```diff
@@ -145,6 +148,33 @@ def derive_depths(original_markers, additional_constraints=[]):
     return solutions
 
 
+def derive_depths_relaxed(original_markers):
+    """Derive paragraph depth with reduced constraints,
+    including removal of emphasis and disabling of the
+    depth_type_inverses optional rule.
+    """
+
+    deemphasized_markers = []
+    logging.warning("""Could not derive paragraph depths. Retrying
+        with relaxed constraints.""")
+    for marker in original_markers:
+        emphasized = re.match(r"<E .*>(.*)</E>", marker)
+        if emphasized:
+            deemphasized_markers.append(emphasized.groups()[0])
+        else:
+            deemphasized_markers.append(marker)
+
+    solutions = derive_depths(deemphasized_markers,
+                              [optional_rules.star_new_level,
+                               optional_rules.limit_paragraph_types(
+                                markers.lower, markers.upper, markers.ints,
+                                markers.roman, markers.em_ints,
+                                markers.em_roman, markers.stars,
+                                markers.markerless)])
```

Comment:
You probably want to move this into the ParagraphProcessor, and remove the hard-coded optional rules. `derive_depths` is called by multiple Processors -- you're just trying to modify the actions of the Regtext version. Perhaps the Regtext processor has a list of fallback optional rules?